### PR TITLE
Reduce compiler warnings in the eclipse project

### PR DESCRIPTION
**Pull request by ernstblechaPT**
This tries to reduce the amount of warnings that is generated for the 4diac IDE plugins in Eclipse.

### CODE REVIEW

**Comment by ernstblechaPT** [comment]
The changes are split into smaller commits to make them easier to understand and discuss,

The commits should probably be squashed when merging.

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/figures/FBNetworkConnection.java [hunk 1]
```diff
@@ -298,8 +298,8 @@ int getMaxFanOutLabelWidth() {
 		final InterfaceEditPart source = (InterfaceEditPart) connEP.getSource();
 		return ((List<ConnectionEditPart>) source.getSourceConnections()).stream()
 				.filter(conn -> !conn.getModel().isVisible())
-				.map(ep -> ep.getFigure().getSourceDecoration().getLabel().getBounds().width).max(Integer::compare)
-				.orElse(0);
+				.map(ep -> Integer.valueOf(ep.getFigure().getSourceDecoration().getLabel().getBounds().width))
+				.max(Integer::compare).orElse(Integer.valueOf(0)).intValue();
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/figures/FBNetworkConnection.java]: Please use an `IntStream` here:
```suggestion
				.mapToInt(ep -> ep.getFigure().getSourceDecoration().getLabel().getBounds().width).max()
				.orElse(0);
```

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/utilities/IsNoReadOnlyEditorTester.java [hunk 2]
```diff
@@ -40,6 +40,7 @@ public boolean test(final Object receiver, final String property, final Object[]
 		return true;
 	}
 
+	@SuppressWarnings("unchecked")
 	private static List<Object> getList(final Object reciever) {
 		if (reciever instanceof final List list) {
 			return list;
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/utilities/IsNoReadOnlyEditorTester.java]:
```suggestion
	private static List<Object> getList(final Object reciever) {
		if (reciever instanceof final List<?> list) {
			return Collections.unmodifiableList(list);
```

File: plugins/org.eclipse.fordiac.ide.deployment.iec61499/src/org/eclipse/fordiac/ide/deployment/iec61499/executors/DynamicTypeLoadDeploymentExecutor.java [hunk 3]
```diff
@@ -408,7 +408,7 @@ private void addFBNetworkElements(final Resource res, final Response object) {
 				} else {
 					fbcmd = new FBCreateCommand(entry, res.getFBNetwork(), 100 * i, 10);
 				}
-				if (fbcmd != null && fbcmd.canExecute()) {
+				if (fbcmd.canExecute()) {
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.deployment.iec61499/src/org/eclipse/fordiac/ide/deployment/iec61499/executors/DynamicTypeLoadDeploymentExecutor.java:411]: I suggest making `fbcmd` `final` and removing the `null` assignment above to ensure a non-`null` value here regardless of future changes.

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editors/FBTypeEditor.java [hunk 4]
```diff
@@ -516,6 +517,7 @@ public void handleContentOutlineSelection(final ISelection selection) {
 		}
 	}
 
+	@SuppressWarnings("static-method")
 	public IEditorActionBarContributor getActionBarContributor() {
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editors/FBTypeEditor.java]: I believe this method is entirely unused.

File: plugins/org.eclipse.fordiac.ide.model.eval/src/org/eclipse/fordiac/ide/model/eval/function/StandardFunctions.java [hunk 5]
```diff
@@ -485,12 +485,12 @@ static <T extends AnyWCharsValue> AnyIntValue FIND(@Comment("The WString to be s
 	/* Array functions */
 
 	@Comment("Returns the lower bound of the dimension of the given array")
-	static <T extends ArrayValue, U extends AnyIntValue> AnyIntValue LOWER_BOUND(final T array, final U dimension) {
+	static <U extends AnyIntValue> AnyIntValue LOWER_BOUND(final ArrayValue array, final U dimension) {
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model.eval/src/org/eclipse/fordiac/ide/model/eval/function/StandardFunctions.java]: The exact signature of this method, including the generic parameters, is important for deriving ST declarations.

File: plugins/org.eclipse.fordiac.ide.model.eval/src/org/eclipse/fordiac/ide/model/eval/function/StandardFunctions.java [hunk 6]
```diff
@@ -485,12 +485,12 @@ static <T extends AnyWCharsValue> AnyIntValue FIND(@Comment("The WString to be s
...
 		return DIntValue.toDIntValue(array.getType().getSubranges().get(dimension.intValue() - 1).getLowerLimit());
 	}
 
 	@Comment("Returns the upper bound of the dimension of the given array")
-	static <T extends ArrayValue, U extends AnyIntValue> AnyIntValue UPPER_BOUND(final T array, final U dimension) {
+	static <U extends AnyIntValue> AnyIntValue UPPER_BOUND(final ArrayValue array, final U dimension) {
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model.eval/src/org/eclipse/fordiac/ide/model/eval/function/StandardFunctions.java]: The exact signature of this method, including the generic parameters, is important for deriving ST declarations.

**Review by mx990 — APPROVED** [review]

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java [hunk 7]
```diff
@@ -48,7 +46,7 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 		}
 
 		cmdstack.execute(getCommand(selected));
-		return Status.OK_STATUS;
+		return null;
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java:49]: Status.OK_STATUS would have been correct please bring back

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java [hunk 8]
```diff
@@ -36,7 +34,7 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 		final IStructuredSelection selection = (IStructuredSelection) HandlerUtil.getCurrentSelection(event);
 		final int i = selection.size() - 1;
 		if (i < 0) {
-			return IStatus.ERROR;
+			return null;
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java:37]: returning null from handlers is considered a bad practice. Better return here Status.CANCEL_STATUS

File: plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java [hunk 9]
```diff
@@ -36,7 +36,7 @@
 import org.eclipse.fordiac.ide.model.data.WcharType;
 import org.eclipse.fordiac.ide.model.data.WordType;
 
-public abstract class BaseRandom {
+public abstract class BaseRandom implements BaseRandomNextValue {
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java:39]: why is this better then the previous version

**Review by azoitl — CHANGES_REQUESTED** [review]
can please give more information on your suppress warning to info commit. With the current commit message I don't get what this does.

**Comment by ernstblechaPT** [comment]
> can please give more information on your suppress warning to info commit. With the current commit message I don't get what this does.

This moves the warnings like
image
to the info level
image
by setting `Unhandled token in '@SuppressWarnings'` to info
image

File: plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java (hunk 9 above)
Comment (ernstblechaPT) [reply plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java:39]: Honestly: I did not want to just add 17 `@SuppressWarning('static-method')` annotations without looking for an alternative solution

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java (hunk 8 above)
Comment (ernstblechaPT) [reply plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java:37]: The reason I went to `null` was the documentation of the execute-method:
image

File: plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java (hunk 9 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/inputgenerator/BaseRandom.java:39]: I can live with that.

**Comment by azoitl** [comment]
> > can please give more information on your suppress warning to info commit. With the current commit message I don't get what this does.
> 
> This moves the warnings like image to the info level image by setting `Unhandled token in '@SuppressWarnings'` to info image

Ah great. These warnings where always annoying me.  But would this also mean that unused surpress warnings (e.g., rawvalue access) would also now be just info or?

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java (hunk 8 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/handler/NewServiceSequenceHandler.java:37]: Strange! I'm pretty sure that I read somewhere that a handler should return a `Status`. But also the examples in the Eclipse documentation return null. So I should stop replacing in my "clean-ups" to replace the null with Status.

**Comment by ernstblechaPT** [comment]
> > > can please give more information on your suppress warning to info commit. With the current commit message I don't get what this does.
> > 
> > 
> > This moves the warnings like image to the info level image by setting `Unhandled token in '@SuppressWarnings'` to info image
> 
> Ah great. These warnings where always annoying me. But would this also mean that unused surpress warnings (e.g., rawvalue access) would also now be just info or?

this only applies to the "unsupported" case - if the annotation is marked as Unnecessary it will still be a warning

**Comment by azoitl** [comment]
Thx for the clarification. That is great!

**Review by azoitl — APPROVED** [review]

**Pull request merged**